go/vt/sqlparser/ast.go

```diff
@@ -6497,113 +6497,116 @@ const (
-			varName, scope, err := VarScope(strings.Split(colName.Name.val, ".")...)
+			varName, scope, explicit, err := VarScope(strings.Split(colName.Name.val, ".")...)
 			if err != nil {
-				return nil, SetScope_None, err
+				return nil, SetScope_None, false, err
 			}
 			if scope == SetScope_None {
-				return colName, scope, nil
+				return colName, scope, false, nil
 			}
-			return &ColName{Name: ColIdent{val: varName}}, scope, nil
+			return &ColName{Name: ColIdent{val: varName}}, scope, explicit, nil
 		} else {
-			varName, scope, err := VarScope(colName.Name.val)
+			varName, scope, explicit, err := VarScope(colName.Name.val)
 			if err != nil {
-				return nil, SetScope_None, err
+				return nil, SetScope_None, false, err
 			}
 			if scope == SetScope_None {
-				return colName, scope, nil
+				return colName, scope, false, nil
 			}
-			return &ColName{Name: ColIdent{val: varName}}, scope, nil
+			return &ColName{Name: ColIdent{val: varName}}, scope, explicit, nil
 		}
 	} else if colName.Qualifier.Qualifier.IsEmpty() { // Forms are like `@@GLOBAL.x` and `@@SESSION.x`
-		varName, scope, err := VarScope(colName.Qualifier.Name.v, colName.Name.val)
+		varName, scope, explicit, err := VarScope(colName.Qualifier.Name.v, colName.Name.val)
 		if err != nil {
-			return nil, SetScope_None, err
+			return nil, SetScope_None, false, err
 		}
 		if scope == SetScope_None {
-			return colName, scope, nil
+			return colName, scope, false, nil
 		}
-		return &ColName{Name: ColIdent{val: varName}}, scope, nil
+		return &ColName{Name: ColIdent{val: varName}}, scope, explicit, nil
 	} else { // Forms are like `@@GLOBAL.validate_password.length`, which is currently unsupported
-		_, _, err := VarScope(colName.Qualifier.Qualifier.v, colName.Qualifier.Name.v, colName.Name.val)
-		return colName, SetScope_None, err
+		_, _, _, err := VarScope(colName.Qualifier.Qualifier.v, colName.Qualifier.Name.v, colName.Name.val)
+		return colName, SetScope_None, false, err
 	}
 }
 
 // VarScope returns the SetScope of the given name, broken into parts. For example, `@@GLOBAL.sys_var` would become
 // `[]string{"@@GLOBAL", "sys_var"}`. Returns the variable name without any scope specifiers, so the aforementioned
-// variable would simply return "sys_var". `[]string{"@@other_var"}` would return "other_var". If the name parts do not
+// variable would simply return "sys_var". `[]string{"@@other_var"}` would return "other_var". If a scope is not
+// explicitly specified, then the boolean return parameter will be false. If the name parts do not
 // specify a variable (returns SetScope_None), then it is recommended to use the original non-broken string, as this
 // will always only return the last part. `[]string{"my_db", "my_tbl", "my_col"}` will return "my_col" with SetScope_None.
-func VarScope(nameParts ...string) (string, SetScope, error) {
+func VarScope(nameParts ...string) (string, SetScope, bool, error) {
```

Comment:
Maybe a little cleaner to add this as a new field on SetScope rather than as a bool here

Reply:
Good idea. `SetScope` is currently just an alias for a `string`, so we'd need to turn it into a struct. Might not be worth it right this minute, but I'll take a closer look.